tests/testthat/test-validate_yaml.R

```diff
@@ -0,0 +1,240 @@
+#---- Valid yaml ----
+test_that("staged dep yaml without dependencies is valid ", {
+
+  text <- "---
+  downstream_repos:
+
+  upstream_repos:
+
+  current_repo:
+    repo: test/test
+    host: https://github.com
+  "
+  expect_silent(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+})
+
+test_that("staged dep yaml allows additional fields beyond those required", {
+
+  text <- "---
+  downstream_repos:
+    - repo: test2/test
+      host: https://github.com
+      another_field : info
+    - repo: test3/test
+      host: https://github.com
+  upstream_repos:
+    - repo: test4/test
+      host: https://github.com
+  current_repo:
+    repo: test/test
+    host: https://github.com
+    another_field : info
+
+  another_field:
+  "
+  expect_silent(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+})
+
+#---- Invalid yaml ----
+
+test_that("empty staged dep yaml file throws error", {
+
+  text <- "---
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
```

Comment:
You may test with a regex in `expect_error` here and below to make sure it throws the right kind of error.